include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -192,21 +194,26 @@ struct TypeInfo : Base<OrtTypeInfo> {
 };
 
 struct Value : Base<OrtValue> {
-  static Value CreateTensor(const OrtAllocatorInfo* info, void* p_data, size_t p_data_len, const int64_t* shape, size_t shape_len,
+  static Value CreateTensor(const AllocatorInfo& info, void* p_data, size_t p_data_len, const int64_t* shape, size_t shape_len,
                             ONNXTensorElementDataType type);
+  static Value CreateMap(Value& keys, Value& values);
+  static Value CreateSequence(std::vector<Value>& values);
 
   Value(nullptr_t) {}
   explicit Value(OrtValue* p) : Base<OrtValue>{p} {}
 
   bool IsTensor() const;
+  size_t GetCount() const;  // If a non tensor, returns 2 for map and N for sequence, where N is the number of elements
+  Value GetValue(int index, OrtAllocator* allocator) const;
```

Comment:
Should allocator be passed by const ptr?

Reply:
The allocator is not really constant if it's used to allocate memory, so I think this is correct.